Change styles of event.

farellano0 opened the pull request:
## Quick Info

- **Feature**: Change style of event information is displayed.
- **Impact**: Frontend and Backend
- **Affected Components**: index and user's events view.

## What does this change?

- Change the table style to card style to see the event's information. Furthermore, add a responsive display.
- While editing styles, we found these errors such as:
  - Wrong redirections in the routes.
  - Validation of privacy statement and set a default value.
  - Check if the image is representable and resizable.

## Why are you changing that?

Implementing these features because the table doesn't match with the application's style.

## Merge/Deploy Checklist
- [x] Events must be visible on the cards.

## How do you manually test this?

To manually test these changes, follow these steps:
1. Navigate to the application and visualize the events created.

## Screenshots
**Desktop**
![Captura desde 2024-02-13 19-20-27](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team-03-s5a1-administrador-de-eventos/assets/73454963/9166c313-c52b-4157-acce-cd4c403e6915)

**Mobile**
![image](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team-03-s5a1-administrador-de-eventos/assets/73454963/b7a40464-b242-4697-adca-b309a682f0a2)

## Errors
We tried to add a validation for active storage where only accept images on jpeg/png files and when upload another kind of file, we got this error:

![image](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team-03-s5a1-administrador-de-eventos/assets/73454963/3be32780-4246-4fbf-80ff-3778fe8e6629)

farellano0 committed:
Fix some settings

While editing styles, we found this errors such as:
- Wrong redirections in the routes.
- Validation of privacy statement and set a default value.
- Check if the image is representable and resizable.

Co-authored-by: Zama <alexavdstiles@gmail.com>
Co-authored-by: etormasa <hector.meza.141@gmail.com>
Co-authored-by: AlejandroEscobarCampos <ae339640@gmail.com>

farellano0 committed:
Major style improve on events views

Change the table style to card style for see event's information. Furthermore, add responsive display.

Co-authored-by: Zama <alexavdstiles@gmail.com>
Co-authored-by: etormasa <hector.meza.141@gmail.com>
Co-authored-by: AlejandroEscobarCampos <ae339640@gmail.com>

MauricioNC commented on app/assets/stylesheets/base/_reset.scss:
```diff
@@ -9,3 +8,8 @@ body {
+
+h1, h2, h3, h4, h5, h6 {
+  margin: 0;
+}
```
Add an empty line at the end of your code

Zama15 replied:
done

MauricioNC commented on app/assets/stylesheets/components/_cards.scss:
```diff
@@ -0,0 +1,63 @@
+      width: 30%;
+    }
+  }
+}
```
Add an empty line at the end of your code

Zama15 replied:
done

MauricioNC commented on app/assets/stylesheets/layout/_header.scss:
```diff
@@ -19,11 +20,12 @@ header {
+  header {
+    flex-direction: row;
   }
 }
```
Add an empty line at the end of your code

Zama15 replied:
done

MauricioNC commented on app/views/bright_events/index.html.erb:
```diff
@@ -5,42 +5,25 @@
-        </tr>
-      <% end %>      
-    </table>
+  <div class="cards-container">
+    <% @events.each do |event|%>
+      <div class="card">
+        <div class="card-header" style="background-image: url(<%= url_for(event.image) if event&.image.attached? %>)">
+        </div>
+        <div class="card-body">
+          <div class="info-event">
+            <h2><%= event.title %></h2>
+            <h3><%= event.user.username %></h3>
+            <h4><%= event.date %></h4>
+            <h5><%= event.location %></h5>
+          </div>
+          <p><%= event.description %></p>
+        </div>
+        <div class="card-footer">
+          <p>$ <%= event.cost %> MX</p>
+          <p><%= event.privacy ? 'Privado' : 'Público' %></p>
+        </div>
+      </div>
+    <% end %>
```
Just as observation, you have almost the same code here and in the user_events.html.erb, may be you can break this in a partial

juperameza replied:
Completely agree!!
I thinks its a must do not a suggection

Zama15 replied:
okay it is not easy but i will look how to deal with it :+1:

Zama15 committed:
Refactor code

user_events and index had the same html structure so was break into a
partial as _cards_container and get call with the render method.
Add h1 in user_events and index to know in what page the users are.

Co-authored-by: etormasa <hector.meza.141@gmail.com>
Co-authored-by: AlejandroEscobarCampos <ae339640@gmail.com>
Co-authored-by: Fernando Arellano <farellano.0709@gmail.com>

Zama15 committed:
Fix error

The gem version active storage have an old version

Co-authored-by: etormasa <hector.meza.141@gmail.com>
Co-authored-by: AlejandroEscobarCampos <ae339640@gmail.com>
Co-authored-by: Fernando Arellano <farellano.0709@gmail.com>
Co-authored-by: juperameza <juperameza@hotmail.com>
